pallets/ocex/src/tests.rs

```diff
@@ -20,113 +20,605 @@ use frame_support::{
+		let mut snapshot = EnclaveSnapshot::<AccountId32, Balance, WithdrawalLimit, AssetsLimit>{
+			snapshot_number: 0,
+    		merkle_root: mmr_root,
+			withdrawals: bounded_vec![],
+    		fees: bounded_vec![],
+
+		};
+		assert_ok!(
+			OCEX::insert_enclave(
+				Origin::root(),
+				account_id.clone().into()
+			)
+		);
+		assert_noop!(
+			OCEX::submit_snapshot(
+				Origin::signed(account_id.into()),
+				snapshot,
+				sig.clone().into()
+			), 
+			Error::<Test>::EnclaveSignatureVerificationFailed
+		);
+	});
+}
+
+#[test]
+fn test_submit_snapshot(){
+	let account_id = create_account_id(); 
+	const PHRASE: &str =
+		"news slush supreme milk chapter athlete soap sausage put clutch what kitten";
+	let public_key_store = KeyStore::new();
+	let public_key = SyncCryptoStore::sr25519_generate_new(
+		&public_key_store,
+		KEY_TYPE,
+		Some(&format!("{}/hunter1", PHRASE)),
+	)
+	.expect("Unable to create sr25519 key pair");
+	let mut t = new_test_ext();
+	t.register_extension(KeystoreExt(Arc::new(public_key_store)));
+	t.execute_with(||{
+		let mmr_root: H256 = create_mmr_with_one_account();
+		let mut snapshot = EnclaveSnapshot::<AccountId32, Balance, WithdrawalLimit, AssetsLimit>{
+			snapshot_number: 0,
+    		merkle_root: mmr_root,
+			withdrawals: bounded_vec![],
+    		fees: bounded_vec![],
+
+		};
+		assert_ok!(
+			OCEX::insert_enclave(
+				Origin::root(),
+				account_id.clone().into()
+			)
+		);
+		let bytes = snapshot.encode();
+		let signature = public_key.sign(KEY_TYPE, &bytes).unwrap();
+		
+		assert_ok!(
+			OCEX::submit_snapshot(
+				Origin::signed(account_id.into()),
+				snapshot,
+				signature.clone().into()
+			),
+		);
+	})
+}
+
+#[test]
+fn test_register_enclave(){
+	let account_id = create_account_id(); 
+	let ias_report = vec![19,19,2,7,255,128,6,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,7,0,0,0,0,0,0,0,7,0,0,0,0,0,0,0,143,217,40,231,107,115,187,6,31,209,73,221,16,254,236,124,82,125,255,243,103,55,60,72,2,212,150,41,217,255,51,13,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,95,183,141,57,75,101,149,246,85,227,219,71,14,143,143,79,2,209,127,165,117,206,185,73,81,228,1,225,150,116,242,38,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,173,228,47,97,157,17,22,141,70,202,241,106,128,71,84,214,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,20,173,193,15,23,193,76,218,39,74,232,62,213,215,194,80];
+	new_test_ext().execute_with(||{
+		assert_ok!(
+			OCEX::register_enclave(
+				Origin::signed(account_id),
+				ias_report
+			)
+		);
+	});
+}
```

Comment:
Tested only positive case. Please add all failing cases and data update/event emitting  assertions.